### PR TITLE
Add DeploymentConflictingPorts analyzer

### PR DESCRIPTION
**Pull request by kebe7jun**
#29302

[X] User Experience
[X] Developer Infrastructure

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: galley/pkg/config/analysis/msg/messages.yaml [hunk 1]
```diff
@@ -371,3 +371,18 @@ messages:
     args:
       - name: annotation
         type: string
+
+  - name: "DeploymentConflictingPorts"
+    code: IST0137
+    level: Warning
+    description: "If we have two services, both selecting the same workload, with the same target port, they should be the same port."
```
Comment (esnible) [inline galley/pkg/config/analysis/msg/messages.yaml:378]: Reword without "if"?  Perhaps "Two services selecting the same workload with same target port are MUST refer to the same port."

File: galley/pkg/config/analysis/msg/messages.yaml [hunk 2]
```diff
@@ -371,3 +371,18 @@ messages:
...
+    template: "This deployment %s is associated with multiple services %v using targetPort %s but different ports: %v."
```
Comment (esnible) [inline galley/pkg/config/analysis/msg/messages.yaml:379]: Nit: use %q instead of %s if port name can be empty.

**Comment by kebe7jun** [comment]
@esnible Thanks, updated.

**Comment by kebe7jun** [comment]
@esnible Hello, any update related to this?

**Comment by kebe7jun** [comment]
@esnible Do i need to do something else?

**Review by esnible — APPROVED** [review]

**Comment by ericvn** [comment]
@kebe7jun @esnible Should these be cherry-picked to release 1.9?

**Comment by esnible** [comment]
@ericvn What do you think, @kebe7jun ?  Usually I do not push hard to get analyzers to cherry-pick, as they are not functional and users can download a daily build of _istioctl_.  If there is 0% chance of the new analyzer incorrectly offering false positive warnings we can cherry pick it.

**Pull request merged**